sensor/README.md

```diff
@@ -1,103 +1,385 @@
+
+If the default project is not enough or does not meet the needs of the user (for example, if they are going to start projects that require running the tools frequently), it is relatively easy to create new projects and calibrator interfaces. In what follows, we will create a new project titled `my_new_project` with a calibrator called `my_new_calibrator`
+
+First, to make sure that the sensor calibration manager knows of the existence of the calibrator, add the following line to `sensor_calibration_manager/sensor_calibration_manager/calibrators/__init__.py`:
+
+```python
+from .my_new_project import *  # noqa: F401, F403
+```
+
+Then, create a file called `sensor_calibration_manager/sensor_calibration_manager/calibrators/my_new_project/__init__.py`
+
+```python
+from .my_new_calibrator import MyNewCalibrator
+
+__all__ = [
+    "MyNewCalibrator",
+]
+```
+
+After this, create a file called `sensor_calibration_manager/sensor_calibration_manager/calibrators/xx1_15/tag_based_pnp_calibrator.py` with the following contents:
+
+```python
+from sensor_calibration_manager.calibrator_base import CalibratorBase
+from sensor_calibration_manager.calibrator_registry import CalibratorRegistry
+from sensor_calibration_manager.ros_interface import RosInterface
+from sensor_calibration_manager.types import FramePair
+
+
+@CalibratorRegistry.register_calibrator(
+    project_name="my_new_project", calibrator_name="my_new_calibrator"
+)
+class MyNewCalibrator(CalibratorBase):
+    required_frames = []
+
+    def __init__(self, ros_interface: RosInterface, **kwargs):
+        super().__init__(ros_interface)
+
+        self.source_frame: str = kwargs["source_frame"]
+        self.target_frame: str = kwargs["target_frame"]
+
+        self.required_frames.extend([self.base_frame, self.source_frame, self.target_frame])
+
+        self.add_calibrator(
+            service_name="calibrate_service_name",
+            expected_calibration_frames=[
+                FramePair(parent=self.target_frame, child=self.source_frame),
+            ],
+        )
+
+```
+
+This calibrator will expect a calibrator between `source_frame` and `target_frame`, which are parameters provided by the user through the launcher file.
+
+Finally, the user must create the correspondent launch file `sensor_calibration_manager/launch/my_new_project/my_new_calibrator.launch.xml`.
+The launcher itself must contain the following arguments to match the calibrator interface:
+
+```xml
+<arg name="source_frame"/>
+<arg name="target_frame"/>
+```
+
+Default values are optional, and somewhere inside the launch file the user needs to execute a node that provides the `calibrate_service_name` to match the calibrator interface
+
+### Integrate a new calibrator
+
+In the previous Section we created a new calibrator interface and added it to the sensor calibration manager. However, that assumes that the user will use one of the calibrator packages offered by the sensor calibration tools.
```

Comment:
Section -> section

Reply:
Usually Section, Chapter, Figure and so on go in capital letters

Reply:
I see! Thanks!